File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.py
def find_result(x):
    if x % 2 == 0:
        return x + 1

    target = f'0{bin(x)[2:]}'
    idx = target.rindex('0')
    result = f'{target[:idx]}10{target[idx+2:]}'
    
    return int(result, 2)


def solution(numbers):    
    return [find_result(n) for n in numbers]

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.java
import java.util.*;

public class Solution {

    public static List<Long> solution(long[] numbers) {
        List<Long> answer = new ArrayList<>();
        for (long number : numbers) {
            answer.add(findResult(number));
        }
        return answer;
    }

    private static long findResult(long x) {
        if (x % 2 == 0) {
            return x + 1;
        }

        String binary = "0" + Long.toBinaryString(x);
        int idx = binary.lastIndexOf('0');

        StringBuilder sb = new StringBuilder(binary);
        sb.setCharAt(idx, '1');
        sb.setCharAt(idx + 1, '0');

        return Long.parseLong(sb.toString(), 2);
    }
}